integration-tests/src/test/scala/za/co/absa/spline/CodelessInitSpec.scala

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright 2017 ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.spline
+
+import java.util.concurrent.atomic.AtomicInteger
+
+import javax.servlet.http.{HttpServlet, HttpServletRequest, HttpServletResponse}
+import org.apache.spark.sql.types.{IntegerType, StringType, StructField, StructType}
+import org.apache.spark.sql.{Row, SaveMode, SparkSession}
+import org.scalatest._
+import org.spark_project.jetty.server.Server
+import org.spark_project.jetty.servlet.ServletHandler
+import za.co.absa.spline.CountingServlet._
+import za.co.absa.spline.harvester.LineageDispatcher.publishUrlProperty
+import za.co.absa.spline.harvester.SparkLineageInitializer.sparkQueryExecutionListenersKey
+import za.co.absa.spline.harvester.listener.SplineQueryExecutionListener
+
+import scala.util.Random
+
+class CodelessInitSpec extends FlatSpec with Matchers with BeforeAndAfterEach with OneInstancePerTest {
```

Comment:
Name `CodelessInitSpec` suggests it's all about codeless initialization stuff. Why should we care about REST, RDD, Schema and the other unrelated things?
Can't it be a normal unit test?

Reply:
For this end2end test to be converted to unit test we would need to provide some way to mock LineageDispatcher via configuration properties e.g. something like what we had for persistence factories.